### PR TITLE
Functional legacy purchase tickets with css module

### PR DESCRIPTION
**Pull request by bgptr**
This diff refactors all components in `components/views/TicketsPage/PurchaseTab/LEGACY_PurchasePage/` and `LEGACY_StakePoolSelect.js` input using hooks and CSS modules.

It ditches `purchaseTickets` connector.
 It ditches `PurchaseTickets.less`,`MiscComponents.less` and `StakePool.less` which is related to #3259. I cherry-picked `ImportScriptIconButton` and `TicketsCogs` buttons from my #3275 to be able to perform this.

Also, it includes automated tests.
 Part of #2438 & #2439.

### CODE REVIEW

File: app/components/views/TicketsPage/PurchaseTab/LEGACY_PurchasePage/LEGACY_AddForm/AddForm.jsx [hunk 1]
```diff
@@ -6,9 +6,7 @@ import {
 } from "buttons";
 import { FormattedMessage as T, injectIntl, defineMessages } from "react-intl";
 import { TextInput, LEGACY_StakePoolSelect } from "inputs";
-import "style/Layout.less";
-import "style/StakePool.less";
-import styles from "../PurchaseTab.module.css";
+import styles from "./AddForm.module.css";
```
Comment (amassarwi) [inline app/components/views/TicketsPage/PurchaseTab/LEGACY_PurchasePage/LEGACY_AddForm/AddForm.jsx:9]: ❤️

File: app/components/views/SettingsPage/SettingsTab/Settings.jsx [hunk 2]
```diff
@@ -15,7 +15,6 @@ import UISettings from "./UISettings";
 import MiscSettings from "./MiscSettings";
 import TimezoneSettings from "./TimezoneSettings";
 import { Subtitle } from "shared";
-import "style/StakePool.less"; // TODO: delete this as well!
```
Comment (amassarwi) [inline app/components/views/SettingsPage/SettingsTab/Settings.jsx:18]: ❤️

File: app/components/views/TicketsPage/PurchaseTab/PurchaseTab.jsx [hunk 3]
```diff
@@ -12,12 +9,10 @@ function Purchase() {
   const [isShowingVsp, toggleShowVsp] = useState(false);
   // const [isShowingImportScript, setShowImportScript] = useState(false);
```
Comment (amassarwi) [inline app/components/views/TicketsPage/PurchaseTab/PurchaseTab.jsx:10]: comment

File: app/components/views/TicketsPage/PurchaseTab/LEGACY_PurchasePage/index.js [hunk 4]
```diff
@@ -8,6 +8,7 @@ import { ExternalLink } from "shared";
 import { EXTERNALREQUEST_STAKEPOOL_LISTING } from "main_dev/externalRequests";
```
Comment (amassarwi) [inline app/components/views/TicketsPage/PurchaseTab/LEGACY_PurchasePage/index.js:8]: Can we rename this `index.js` to something more meaningful ? and use `index.js` to export it properly.

File: app/components/views/TicketsPage/PurchaseTab/PurchaseTickets/PurchaseForm.jsx [hunk 5]
```diff
@@ -4,8 +4,6 @@ import { PassphraseModalButton, KeyBlueButton } from "buttons";
 import { AccountsSelect, NumTicketsInput, VSPSelect } from "inputs";
 import { Balance } from "shared";
 import styles from "../PurchaseTab.module.css";
-// XXX this should go away!
-import "style/StakePool.less";
```
Comment (amassarwi) [inline app/components/views/TicketsPage/PurchaseTab/PurchaseTickets/PurchaseForm.jsx:8]: 😻

**Review by amassarwi — APPROVED** [review]
Nice work - left one suggestion.

File: app/components/views/TicketsPage/PurchaseTab/PurchaseTickets/PurchaseForm.jsx [hunk 6]
```diff
@@ -4,8 +4,6 @@ import { PassphraseModalButton, KeyBlueButton } from "buttons";
 import { AccountsSelect, NumTicketsInput, VSPSelect } from "inputs";
 import { Balance } from "shared";
 import styles from "../PurchaseTab.module.css";
```
Comment (amassarwi) [inline app/components/views/TicketsPage/PurchaseTab/PurchaseTickets/PurchaseForm.jsx:6]: Just noting that we should avoid such css modules imports, instead this component should have it's own encapsulated css module.

When we are done with #2439 I'm gonna create a follow up issue listing all components with such problematic import statements.

Comment (bgptr) [reply app/components/views/TicketsPage/PurchaseTab/PurchaseTickets/PurchaseForm.jsx:6]: You are right. I noticed it, but I did not want to deal with it in this PR. Although, I strived for creating separate css modules in the legacy ticket purchase part.

Comment (amassarwi) [reply app/components/views/TicketsPage/PurchaseTab/PurchaseTickets/PurchaseForm.jsx:6]: Yep, thanks for that 👍

**Comment by bgptr** [comment]
@amassarwi thanks for your review. I've fixed your suggestions.

**Comment by bgptr** [comment]
@amassarwi rebased

**Review by amassarwi — APPROVED** [review]
@bgptr thanks for ditching `MiscComponents.less`.  LGTM.

**Review by alexlyp — APPROVED** [review]

**Pull request merged**